query/src/main/java/org/infinispan/query/CacheQuery.java

```diff
@@ -103,9 +103,9 @@
     * If the projected field is not a projectable field, null is returned in the object[]
     *
     * @param fields the projected field names
-    * @return {@code this}  to allow for method chaining
+    * @return {@code this} to allow for method chaining, but the type parameter now becomes {@code Object[]}
     */
-   CacheQuery projection(String... fields);
+   CacheQuery<Object[]> projection(String... fields);
```

Comment:
This is the only think that may break compatibility, but I prefer this version with ```Object[]```